allennlp/data/vocabulary.py

```diff
@@ -251,7 +251,7 @@ def set_from_file(self,
         with codecs.open(filename, 'r', 'utf-8') as input_file:
             for i, line in enumerate(input_file.readlines()):
                 index = i + 1 if is_padded else i
-                token = line.strip()
```

Comment:
Because if you have a unicode non-breaking space character, it gets an index, but gets stripped here.  I did this because I was getting errors reading a vocab file - the number of read vocab items didn't match the saved embedding dimension.  I'm guessing spacy doesn't split on some unicode characters that my tokenizer did.

We're assuming the vocab was written by our code, and that python handles line endings correctly, so this should always be correct.